### PR TITLE
Reorganize Iceberg connector documentation

### PR DESCRIPTION
**Pull request by m57lyra**
## Description

**THERE ARE NO TECHNICAL CONTENT CHANGES TO EXISTING MATERIAL IN THIS PR**

This PR is a follow-on to PR#15637. It reorganizes the Iceberg connector documentation to match the structure of the Hive connector documentation and, as with the Hive PR, does the following:

- Fixes problematic verb tenses as per Google Developer Style Guide (must use present tense only).
  These included would, should, could, needs to, will be; there were not many instances.
- Formats the few tables remaining in the old format to the list table format (and of course adds table titles). 
- Adds intro blurbs where there were none for the following section headers:
  - Metastores
  - Security
  - Procedures (used same language as the Hive blurb)
  - Table properties (used same language as the Hive blurb)
  - View management (used same language as the Hive blurb)
  - Performance  (used same language as the Hive blurb)
  - File formats  (used same language as the Hive blurb)

## Additional context and related issues

Here is a possibly helpful screen cap showing how the Iceberg doc org lines up with the Hive doc org now:

Screenshot 2023-03-08 at 3 59 33 PM

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 1]
```diff
@@ -233,408 +122,121 @@ is used.
...
-You can enable authorization checks for the connector by setting
-the ``iceberg.security`` property in the catalog properties file. This
-property must be one of the following values:
+Metastores
+----------
 
-.. list-table:: Iceberg security values
-  :widths: 30, 60
-  :header-rows: 1
+Iceberg natively uses the concept of a catalog. The subsection titles in this
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:128]: How about this as a replacement for the whole paragraph.

The Iceberg table format manages most metadata in metadata files in the object storage itself. A small amount of metadata however still requires the use of a metastore. In the Iceberg ecosystem, these smaller metastores are called Iceberg metadata catalogs, or just catalogs. The following sections detail the available options of metadata catalogs you can use.

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 2]
```diff
@@ -233,408 +122,121 @@ is used.
...
 
-The following table properties can be updated after a table is created:
+Hive metastore catalog
+^^^^^^^^^^^^^^^^^^^^^^
 
-* ``format``
-* ``format_version``
-* ``partitioning``
-* ``sorted_by``
+The Hive metastore catalog is the default implementation.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:143]:
```suggestion
The Hive metastore catalog is the default.
```

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 3]
```diff
@@ -233,408 +122,121 @@ is used.
...
+When using it, the Iceberg connector supports the same metastore
+configuration properties as the Hive connector. At a minimum,
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:145]: I assume we will rejig this later ...

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 4]
```diff
@@ -233,408 +122,121 @@ is used.
...
-      allowed.
-  * - ``FILE``
-    - Authorization checks are enforced using a catalog-level access control
-      configuration file whose path is specified in the ``security.config-file``
-      catalog configuration property. See
-      :ref:`catalog-file-based-access-control` for information on the
-      authorization configuration file.
+The connector supports multiple Iceberg catalog types; you may use
+either a Hive metastore service (HMS), AWS Glue, or a REST catalog. The catalog
+type is determined by the ``iceberg.catalog.type`` property. It can be set to
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:135]: Maybe the values should be in each section ..

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 5]
```diff
@@ -233,408 +122,121 @@ is used.
...
+When using the Glue catalog, the Iceberg connector supports the same
+configuration properties as the Hive connector's Glue setup. See
+:ref:`AWS Glue metastore configuration<hive-glue-metastore>`.
 
-    ALTER TABLE table_name SET PROPERTIES partitioning = ARRAY[<existing partition columns>, 'my_new_partition_column'];
+.. code-block:: text
 
-The current values of a table's properties can be shown using :doc:`SHOW CREATE TABLE </sql/show-create-table>`.
+    connector.name=iceberg
+    iceberg.catalog.type=glue
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:166]: so this is not sufficient as a minimal example then right?

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 6]
```diff
@@ -233,408 +122,121 @@ is used.
...
+.. _iceberg-rest-catalog:
 
-The Iceberg connector supports setting comments on the following objects:
+REST catalog
+^^^^^^^^^^^^^^
 
-- tables
-- views
-- table columns
+In order to use the Iceberg REST catalog, ensure to configure the catalog type with
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:173]: This is inconsistent with the other two before.. we should rejig

Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.rst:173]: Also .. we recently added some info about limitation for JDBC and REST.. have those gone lost? Or did I just not see it or misremember

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 7]
```diff
@@ -233,408 +122,121 @@ is used.
...
-from :ref:`Partitioned Tables <iceberg-tables>` section,
-the following SQL statement deletes all partitions for which ``country`` is ``US``::
+  The JDBC catalog may face the compatibility issue if Iceberg introduces breaking changes in the future.
+  Consider the :ref:`REST catalog <iceberg-rest-catalog>` as an alternative solution.
 
-    DELETE FROM example.testdb.customer_orders
-    WHERE country = 'US'
+At a minimum, ``iceberg.jdbc-catalog.driver-class``, ``iceberg.jdbc-catalog.connection-url`` and
+``iceberg.jdbc-catalog.catalog-name`` must be configured.
+When using any database besides PostgreSQL, a JDBC driver jar file must be placed in the plugin directory.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:225]: JAR file

Comment (mosabua) [reply docs/src/main/sphinx/connector/iceberg.rst:225]: and trino-iceberg plugin directory

**Review by mosabua — COMMENTED** [review]
First part of a review done .. I will need to revisit and also run this locally to see.

File: docs/src/main/sphinx/connector/iceberg.rst (hunk 3 above)
Comment (m57lyra) [reply docs/src/main/sphinx/connector/iceberg.rst:145]: I'm not touching any existing content as per the agreement. Only filling in missing intros.

File: docs/src/main/sphinx/connector/iceberg.rst (hunk 5 above)
Comment (m57lyra) [reply docs/src/main/sphinx/connector/iceberg.rst:166]: This is existing content. Not touching it in this PR as per agreement.

File: docs/src/main/sphinx/connector/iceberg.rst (hunk 6 above)
Comment (m57lyra) [reply docs/src/main/sphinx/connector/iceberg.rst:173]: This is existing content, which we promised not to change. This is only a reorganization of what currently exists. If there are changes to content they have to be done in a separate PR.

File: docs/src/main/sphinx/connector/iceberg.rst (hunk 7 above)
Comment (m57lyra) [reply docs/src/main/sphinx/connector/iceberg.rst:225]: This is existing content. This PR won't change existing content, only reorder it as per agreement with maintainers.

File: docs/src/main/sphinx/connector/iceberg.rst (hunk 4 above)
Comment (m57lyra) [reply docs/src/main/sphinx/connector/iceberg.rst:135]: That has to be done in a separate PR as it introduces changes to existing technical content.

File: docs/src/main/sphinx/connector/iceberg.rst (hunk 2 above)
Comment (m57lyra) [reply docs/src/main/sphinx/connector/iceberg.rst:143]: This is existing technical content and outside the scope of the PR.

**Comment by m57lyra** [comment]
Pushed verb tense fixes since review is delayed so this can move along faster.

File: docs/src/main/sphinx/connector/iceberg.rst (hunk 1 above)
Comment (m57lyra) [reply docs/src/main/sphinx/connector/iceberg.rst:128]: Met you half way.

**Comment by m57lyra** [comment]
Includes Manfred's feedback that does not include existing content.

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 8]
```diff
@@ -749,226 +353,396 @@ this table:
...
     - ``MAP(k,v)``
 
-No other types are supported.
+No other types are supported.
+
+Security
+--------
+
+The Iceberg connector allows you to choose one of several means of providing 
+autorization at the catalog level. You can select a different type of
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:362]:
```suggestion
authorization at the catalog level.
```

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 9]
```diff
@@ -749,226 +353,396 @@ this table:
...
 
-No other types are supported.
+No other types are supported.
+
+Security
+--------
+
+The Iceberg connector allows you to choose one of several means of providing 
+autorization at the catalog level. You can select a different type of 
+authorization check in different Iceberg catalog files.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:363]:
```suggestion

```

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 10]
```diff
@@ -749,226 +353,396 @@ this table:
...
+    )
+    AS VALUES
+        (2021, 10000),
+        (2022, 20000);
+
+Procedures
+^^^^^^^^^^
+
+Use the :doc:`/sql/call` statement to perform data manipulation or
+administrative tasks. Procedures must include a qualified catalog name, if your
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:493]: Update as suggested in the Delta Lake PR

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 11]
```diff
@@ -1419,5 +1387,78 @@ specify a subset of columns to analyzed with the optional ``columns`` property::
...
+
+The connector supports redirection from Iceberg tables to Hive tables
+with the ``iceberg.hive-catalog-name`` catalog configuration property.
+
+File formats
+------------
+
+The following file types and formats are supported for the Iceberg connector:
+
+* ORC
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:1408]: Really .. all of these are supported in Iceberg .. I thought for data files just ORC and Parquet

**Review by mosabua — CHANGES_REQUESTED** [review]

**Comment by m57lyra** [comment]
Closing in favor of https://github.com/trinodb/trino/pull/16835

**Pull request closed without merging**